security/pkg/pki/ra/k8s_ra_test.go

```diff
@@ -92,14 +92,19 @@ func TestK8sSignWithMeshConfig(t *testing.T) {
 			updatedRootCertForMeshConfig: TestCACertFile,
 			expectedFail:                 true,
 		},
+		{
+			name:      "Root cert is not specified in mesh config and cert chain contains only intermediate CA(only leaf cert + intermediate CA)",
```

Comment:
Should these tests be added in a separate test function called `TestK8sSignWithCertChain`?

Reply:
We can just add a TC in this function to test that  the cert in certchain is not a CA cert

Reply:
Does the root certificate have to be included in either the MeshConfig or the cert chain or is it sufficient to have only the intermediate certificate in the cert chain (and in this case be considered a root cert)? If the intermediate cert cannot be considered a root, should this method return an error if a root certificate isn't found in either location? From  the method description:
```go
If no root cert can be found from either of the two sources, error returned.
```
I think this test scenario should be included since none of the previous cases tested the case where the root cert is not specified in the MeshConfig, but an intermediate cert is included in the cert chain. Is it possible for this configuration to occur?

Reply:
The current test cases are focuing on getting root cert from mesh config, as root cert from cert chain is the default behaviour.  Yep, it will be helpful if you can add more cases to root cert from cert chain only.